src/lib/entropy/darwin_secrandom/darwin_secrandom.cpp

```diff
@@ -0,0 +1,28 @@
+/*
+* Darwin SecRandomCopyBytes EntropySource
+* (C) 2015 Daniel Seither (Kullo GmbH)
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/internal/darwin_secrandom.h>
+#include <Security/Security.h>
+
+namespace Botan {
+
+/**
+* Gather entropy from SecRandomCopyBytes
+*/
+void Darwin_SecRandom::poll(Entropy_Accumulator& accum)
+   {
+   const size_t ENTROPY_BITS_PER_BYTE = 8;
+   const size_t BUF_SIZE = 256;
+
+   m_buf.resize(BUF_SIZE);
+   if (0 == SecRandomCopyBytes(kSecRandomDefault, m_buf.size(), m_buf.data()))
```

Comment:
This depends on the general expectation of what an entropy source should do if it fails to provide new entropy.

I frankly don't know the failure modes of [SecRandomCopyBytes](https://developer.apple.com/library/ios/documentation/Security/Reference/RandomizationReference/index.html#//apple_ref/c/func/SecRandomCopyBytes), as they are not documented. The only thing I know is that `errno` is set in case of an error, which indicates that it occurred while doing a syscall or the like; probably while reading /dev/random.

When the dev_random Botan module fails to read data, it simply returns without throwing, so this is probably the way to go.